sdk/core/azure-core/test/e2e/azure_hang.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief Performs  upload and download for n times trying to catch long running bugs.
+ * Originally set to use 50Mb for 2000 times, which takes ~2.5hours to complete.
```

Comment:
That's quite a long time. How frequently do these tests run? Do we really need to run it 2000 times/for 2.5 hours to verify there is no hang?

Even if this is running once a week, we should try to bring the test time down to < 10 minutes, either by doing more things in parallel and/or reducing the size/iteration count to under 100.

Reply:
We are currently not running e2e tests from CORE
This tests are there to be manually run whenever we want to test or debug something.
This code, for example, was provided by @Jinming-Hu  and I added as an e2e test we can continue to be used in the future if we want.

Reply:
I don't know how I feel about tests/code that we don't intend to run to the repo. Does this at least get compiled/built?